shopelectro/management/commands/price.py

```diff
@@ -129,36 +158,63 @@ def filter_categories(utm):
         def prepare_products(categories_, utm):
             """Filter product list and patch it for rendering."""
             products = PriceFilter(utm).run(
-                Product.objects.active().filter_by_categories(categories_)
+                models.Product.objects.active().filter_by_categories(categories_)
             )
-            brands = Tag.objects.get_brands(products)
+            brands = models.Tag.objects.get_brands(products)
             return [
                 put_brand(put_crumbs(put_utm(product)), brands)
                 for product in products
             ]
 
         categories = (
-            filter_categories(utm) if utm != 'SE78'
-            else Category.objects.all()
+            filter_categories(self.target) if self.target != 'SE78'
+            else models.Category.objects.all()
         )
 
-        products = prepare_products(categories, utm)
+        products = prepare_products(categories, self.target)
 
         return {
             'base_url': settings.BASE_URL,
             'categories': categories,
             'products': products,
             'shop': settings.SHOP,
-            'utm': utm,
+            'utm': self.target,
         }
 
-    @classmethod
-    def generate_yml(cls, utm, file_name):
-        """Generate yml file."""
-        file_to_write = os.path.join(cls.BASE_DIR, file_name)
-        context = cls.get_context_for_yml(utm)
 
-        with open(file_to_write, 'w', encoding='utf-8') as file:
-            file.write(render_to_string('prices/price.yml', context).strip())
+class Category:
+    pass
+
+
+class Categories:
+    pass
+
+
+class Product:
+    pass
+
 
-        return '{} generated...'.format(file_name)
+class Products:
+    pass
+
+
+# --- command block ---
+class Command(BaseCommand):
+    """Generate yml file for a given vendor (YM or price.ru)."""
+
+    # price files will be stored at this dir
+    BASE_DIR = settings.ASSETS_DIR
+
+    IGNORED_CATEGORIES = [
+        'Измерительные приборы', 'Новогодние вращающиеся светодиодные лампы',
+        'Новогодние лазерные проекторы', 'MP3- колонки', 'Беспроводные звонки',
+        'Радиоприёмники', 'Фонари', 'Отвертки', 'Весы электронные портативные',
+    ]
+
+    # dict keys are url targets for every service
+    IGNORED_CATEGORIES_MAP = defaultdict(list, {
+        'GM': ['Усилители звука для слабослышащих'],
+    })
+
+    def handle(self, *args, **options):
+        Files(settings.UTM_PRICE_MAP, self.BASE_DIR).create()
```

Comment:
I believe it will be more more of OOP like solution.
- It represents a relationship between objects
- It allows us to see which data the object owns and handles
- Objects contain only related data

```suggestion
        Files([
            File(
                path=join(self.BASE_DIR, file_name), 
                target=Target(target_name)
            ) for target_name, file_name in self.utm_price_map.items()
        ]).create()
```

Reply:
@duker33 what about this comment? It would be nice to create a todo for it or implement it in this  PR?

Reply:
@artemiy312 , sorry, forgot about it.
I'll fix it at the same task